src/Adapter/Requirement/CheckRequirements.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Requirement;
+
+use ConfigurationTest;
+use Symfony\Component\Translation\TranslatorInterface;
+
+
+/**
+ * Check system requirements of a PrestaShop website
+ */
+class CheckRequirements
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * Returns a summary of all system requirements
+     *
+     * @return array
+     */
+    public function getSummary()
+    {
+        $paramsRequiredResults = ConfigurationTest::check(ConfigurationTest::getDefaultTests());
+
+        $isHostMode = defined('_PS_HOST_MODE_');
+
+        if (!$isHostMode) {
+            $paramsOptionalResults = ConfigurationTest::check(ConfigurationTest::getDefaultTestsOp());
+        }
+
+        $failRequired = in_array('fail', $paramsRequiredResults);
+
+        $testsErrors = $this->getErrorMessages();
+
+        if ($failRequired && $paramsRequiredResults['files'] != 'ok') {
```

Comment:
Please use a strict check and apply Yoda conditions here
`if ($failRequired && 'ok' !== $paramsRequiredResults['files']) {`
You can find it [here](https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md#comparison)